src/browser-sandbox/pruneNonCriticalSelectors.js

```diff
@@ -83,19 +83,25 @@ export default function pruneNonCriticalSelectors ({
       }
     })
   }
+
   // not using timeout because does not work with JS disabled
   function sleep (time) {
     return new Promise(resolve =>
       pollUntilTimePassed(Date.now(), time, resolve)
     )
   }
 
-  // give some time (renderWaitTime) for sites like facebook that build their page dynamically,
-  // otherwise we can miss some selectors (and therefor rules)
-  // --tradeoff here: if site is too slow with dynamic content,
-  // it doesn't deserve to be in critical path.
-  return sleep(renderWaitTime).then(() => {
-    console.log('debug: waited for renderWaitTime: ' + renderWaitTime)
-    return filterSelectors(selectors)
-  })
+  try {
```

Comment:
Did you test putting this try/catch inside core.js instead?  As long as that still catches the error, I prefer it, as we can react to the error, instead of just receiving no selectors from this function. I implemented this in [my branch here](https://github.com/pocketjoso/penthouse/pull/228/files#diff-7eb52b366866677666470e019283c8eaR177), but I don't have an easy way to test the issues you have been testing, so I don't know 100% that it catches your errors, but it looks like it should.

Reply:
The reason for me to do it that way was to ensure to solve the execution context problem. But I tried a lot of things. So I will check if its running safe with the try catch in core. In common you are right.

Reply:
tested and merged this in (via core.js) in #228